dash_cytoscape/utils/Tree.py

```diff
@@ -0,0 +1,160 @@
+            self.edge_data = {}
+
+    def _dfs(self, search_id):
+        if self.node_id == search_id:
+            return self
+        elif self.is_leaf():
+            return None
+        else:
+            for child in self.children:
+                result = child.dfs()
+                if result:
+                    return result
+
+            return None
+
+    def _bfs(self, search_id):
+        stack = deque([self])
+
+        while stack:
+            tree = stack.popleft()
+
+            if tree.node_id == search_id:
+                return tree
+
+            if not tree.is_leaf():
+                for child in tree.children:
+                    stack.append(child)
+
+        return None
+
+    def is_leaf(self):
+        """
+        :return: If the Tree is a leaf or not
+        """
+        return not self.children
+
+    def add_child(self, child):
+        """
+        Add a single child to the children of a Tree
+        :param child: Tree object
+        """
+        self.children.append(child)
+
+    def add_children(self, children):
+        """
+        Add a list of children to the current children of a Tree
+        :param children: List of Tree objects
+        """
+        self.children.extend(children)
+
+    def get_edges(self):
+        """
+        Get all the edges of the tree in Cytoscape JSON format
+        :return: List of dictionaries, each specifying an edge
+        """
+        edges = [
+            {
+                'data': {
+                    'source': self.node_id,
+                    'target': child.node_id,
+                    **self.edge_data
+                }
+            }
+            for child in self.children
+        ]
+
+        for child in self.children:
+            edges.extend(child.get_edges())
+
+        return edges
+
+    def get_nodes(self):
+        """
+        Get all the nodes of the tree in Cytoscape JSON format
+        :return: List of dictionaries, each specifying a node
+        """
+        nodes = [
+            {
+                'data': {
+                    'id': self.node_id,
+                    **self.data
+                }
+            }
+        ]
+
+        for child in self.children:
+            nodes.extend(child.get_nodes())
+
+        return nodes
+
+    def get_elements(self):
+        """
+        Get all the elements of the tree in Cytoscape JSON format
+        :return: List of dictionaries, each specifying an element
+        """
+        return self.get_nodes() + self.get_edges()
+
+    def find_by_id(self, search_id, method='bfs'):
+        """
+        Finds a Tree object by its ID
+        :param search_id: the queried ID
+        :param method: Which traversal method to use. Either "bfs" or "dfs"
+        :return: Tree object if found, None otherwise
+        """
+        for c in ' -_':
+            method = method.replace(c, '')
+        method = method.lower()
+
+        if method in ['bfs', 'breadthfirst', 'breadthfirstsearch']:
```

Comment:
I think if the docstring says `Either "bfs" or "dfs"`, those should be the only two values this function accepts for the `method` parameter.

Reply:
That's a good Idea.